articles/hapi-vs-koa-vs-express/index.md

```diff
@@ -0,0 +1,159 @@
+6. Hapi provides good plugins such as Yar, Good, Catbox etc.
+
+##### Disadvantages
+1. Refactoring in Hapi is manual.
+2. Endpoints must be created and tested manually.
+3. It restricts developers to using Hapi specific plugins and modules which are not compatible with other frameworks.
+
+#### Koa.js
+##### Advantages
+1. Koa has a small footprint hence making it very light weight.
+2. Koa has a good overall user experience.
+3. Koa uses try/catch which improves error handling.
+4. Koa has a cleaner and more readable asynchronous code.
+5. Koa does not use callbacks hence facilitating upstream and downstream flow of controls.
+
+##### Disadvantages
+1. Koa has a relatively small community.
+2. It is not compatible with middleware builds using express style.
+3. Koa uses ES6 generators which are not compatible with middleware from other framework.
+
+#### Express.js
+##### Advantages
+1. Express has a large community.
+2. Express is simple, flexible, and scalable.
+3. Express is fully customizable.
+4. It is easy to integrate third party services and middleware.
+5. Express has a low learning curve.
+
+##### Disadvantages
+1. Express requires a lot of manual labor.
+2. Refactoring bigger code bases is very challenging.
+
+### Framework performance
+In this section, we will compare the frameworks based on the number of request that a hello world server can handle in one second.
+
+Hapi.js handles 29,998 requests per second while Express handles 38,510 requests per second.
+
+Koa.js seems to be best performing since it handles 50,933 requests per second.
+
+### Choosing the right framework for your project
+
+#### Why choose Express.js
+Express provides a wide range of functionalities for developing web applications such as single/multi-page applications and even hybrid applications. Express is also easy to test due to its middleware and routes which modularize projects. Express.js framework is also closer to Node.js when compared to the other Frameworks, it is only a thin layer to Node.js hence making it the best for building applications that require significantly better performance.
+
+#### Why choose Koa.js
+Koa.js was created to be better than express, therefore, it can do all that express can do and more. From our performance test we saw that Koa performs the best since it can handle many requests per second. This makes it a better choice for web applications and APIs that serve many clients at a time.
+
+#### Why choose Hapi.js
+Hapi.js has almost the same capabilities as the other frameworks but its plugin feature is what makes it unique when it comes to selecting frameworks to use. Hapi has a wide variety of plugins that make it a little bit easier to build APIs with since almost every common task such as routing, logging, caching, authentication etc. have all-ready built plugins that have been tested.    
+
+Below are some other points that would help when choosing the right framework for a project.
```

Comment:
You should expand on this section, by providing the best framework for each of these points.

Reply:
Hello, these points depend on the project that the reader has (which is unknown) therefore, there is no specific criteria to determine the best framework for each of the points.